### PR TITLE
[ONNX] Add support for Operator Sets

### PR DESCRIPTION
**Pull request by apwojcik**
This is the initial version of the operators set. The member functions `OperatorsBridge::get_operator_set_X` will be used in later PRs to add ONNX operators specific to a version of operator set.

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/op/average_pool.hpp [hunk 1]
```diff
@@ -26,17 +26,21 @@ namespace ngraph
...
-             *
-             * @return The vector containing Ngraph nodes producing output of ONNX AveragePool
-             *         operation.
-             */
-            NodeVector average_pool(const Node& node);
-
-        } // namespace op
+            namespace set_1
+            {
+                /**
```
[the remark this answers is not in the document]
Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/op/average_pool.hpp]: The comment is not the part of the change, only the addition of the namespace. I prefer to create separate PR to correct this.

File: src/ngraph/frontend/onnx_import/core/model.hpp [hunk 2]
```diff
@@ -30,6 +30,15 @@ namespace ngraph
             explicit Model(const onnx::ModelProto& model_proto)
                 : m_model_proto{&model_proto}
             {
+                for (const auto& id : m_model_proto->opset_import())
```
[the remark this answers is not in the document]
Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/core/model.hpp]: Yes, the underneath type is `const ::google::protobuf::RepeatedPtrField< ::onnx::OperatorSetIdProto >&`

File: src/ngraph/frontend/onnx_import/core/graph.cpp [hunk 3]
```diff
@@ -21,6 +21,19 @@ namespace ngraph
 {
     namespace onnx_import
     {
+        namespace detail
+        {
+            std::string to_string(const std::vector<std::string>& vector)
```
[the remark this answers is not in the document]
Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/core/graph.cpp]: The main usage of `ngraph::vector_to_string` is for numerical elements and it presents the content of vector as `[1, 2, 3,... ]`. I need just a list of strings coma separated, i.e. `abc, defg, hijk, ...`.

[the remark this answers is not in the document]
Comment (arogowie-intel) [reply src/ngraph/frontend/onnx_import/core/graph.cpp]: Ah, I've missed that you don't need brackets.

File: src/ngraph/frontend/onnx_import/ops_bridge.cpp [hunk 4]
```diff
@@ -86,42 +86,194 @@ namespace ngraph
         {
             namespace error
             {
-                struct unknown_operation : ngraph_error
+                struct UnknownOperator : ngraph_error
                 {
-                    explicit unknown_operation(const std::string& op_type)
-                        : ngraph_error{"unknown operation: " + op_type}
+                    explicit UnknownOperator(const std::string& op_type)
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/ops_bridge.cpp]: Please add `<string>` header.

File: src/ngraph/frontend/onnx_import/ops_bridge.hpp [hunk 5]
```diff
@@ -16,17 +16,19 @@
 
 #pragma once
 
-#include "core/node.hpp"
-#include "ngraph/node_vector.hpp"
+#include <string>
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/ops_bridge.hpp]: Unnecessary include in this header.

File: src/ngraph/frontend/onnx_import/ops_bridge.hpp [hunk 6]
```diff
@@ -16,17 +16,19 @@
...
 
 namespace ngraph
 {
     namespace onnx_import
     {
         namespace ops_bridge
         {
-            NodeVector make_ng_nodes(const onnx_import::Node&);
-        }
+            const OperatorSet& get_operator_set(std::int64_t version);
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/ops_bridge.hpp]: Please add `<cstdint>` header.

File: src/ngraph/frontend/onnx_import/ops_bridge.cpp [hunk 7]
```diff
@@ -86,42 +86,194 @@ namespace ngraph
...
+                static const OperatorSet& get_operator_set(std::int64_t version)
                 {
-                    return ops_bridge::get()(node);
+                    return instance().get_operator_set_version(version);
                 }
 
             private:
-                std::map<std::string, std::function<NodeVector(const Node&)>> m_map;
+                std::map<std::string,
+                         std::map<std::int64_t, std::function<NodeVector(const Node&)>>>
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/ops_bridge.cpp]: Please add `<map>` header.

File: src/ngraph/frontend/onnx_import/ops_bridge.cpp [hunk 8]
```diff
@@ -86,42 +86,194 @@ namespace ngraph
...
                 {
-                    static ops_bridge instance;
+                    static OperatorsBridge instance;
                     return instance;
                 }
 
+                const Operator& get_operator(const std::string& name, std::int64_t version) const
+                {
+                    auto op = m_map.find(name);
+                    if (op == std::end(m_map))
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/ops_bridge.cpp]: Please add `<iterator>` header.

File: src/ngraph/frontend/onnx_import/ops_bridge.cpp [hunk 9]
```diff
@@ -86,42 +86,194 @@ namespace ngraph
         {
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/ops_bridge.cpp]: Please remove unused `<algorithm>` header.

File: src/ngraph/frontend/onnx_import/core/operator_set.hpp [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        // Forward declaration
+        class Node;
+
+        using Operator = std::function<NodeVector(const Node&)>;
+        using OperatorSet = std::unordered_map<std::string, std::reference_wrapper<const Operator>>;
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/core/operator_set.hpp]: Please add header `<string>`.

File: src/ngraph/frontend/onnx_import/ops_bridge.cpp (hunk 4 above)
Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/ops_bridge.cpp]: It is already there, check the top of the file.

Comment (arogowie-intel) [reply src/ngraph/frontend/onnx_import/ops_bridge.cpp]: I don't see it there. It is included in `ops_bridge.hpp` where it is unnecessary.

Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/ops_bridge.cpp]: What version of changes are you looking at? None of the commits add/remove the <string> from the file preamble, and it is there.

File: src/ngraph/frontend/onnx_import/ops_bridge.hpp (hunk 6 above)
Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/ops_bridge.hpp]: It is there, again what version of changes are you looking at?

File: src/ngraph/frontend/onnx_import/ops_bridge.cpp (hunk 8 above)
Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/ops_bridge.cpp]: done

File: src/ngraph/frontend/onnx_import/ops_bridge.cpp (hunk 9 above)
Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/ops_bridge.cpp]: done

Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/ops_bridge.cpp]: done

File: src/ngraph/frontend/onnx_import/ops_bridge.cpp (hunk 7 above)
Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/ops_bridge.cpp]: done

File: src/ngraph/frontend/onnx_import/op/average_pool.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (rkimballn1) [reply src/ngraph/frontend/onnx_import/op/average_pool.hpp]: You are here so why not change it now.

[the remark this answers is not in the document]
Comment (apwojcik) [reply src/ngraph/frontend/onnx_import/op/average_pool.hpp]: Done

**Pull request merged**